### DISCUSSION TITLE
Variable Value Callback

### DISCUSSION THREAD

**Discussion by mattiacorpora** [FreeOpcUa/opcua-asyncio#1606] [Q&A]
Hello,
I'm new to using asyncua and I'm finding it quite comprehensive. I have a question regarding variable handling with callbacks. I'm interested in understanding how to set up callbacks for both reading and writing variables in asyncua.
Specifically, I would like to ensure that on the server side, every time a client reads a variable, a callback is triggered before the read operation, and another callback is activated after a write operation.
Could you please provide guidance or examples on how to achieve this using asyncua?
Thank you for your assistance.

**Comment by AiyionPrime** [comment]
@oroulet I was just looking for the same feature.
Is it possible to invoke functions upon OPC UA variable read?

**Comment by oroulet** [comment]
was hard to find back but look at server-callback.py. it allows to instrument a bit the code

**Reply by AiyionPrime** [reply]
Gotta test, whether that's what I had in mind, thanks so far!
